src/System.Text.Json/tests/Serialization/DictionaryTests.cs

```diff
@@ -862,6 +862,130 @@ public static void DeserializeUserDefinedDictionaryThrows()
             Assert.Throws<NotSupportedException>(() => JsonSerializer.Parse<IImmutableDictionaryWrapper>(json));
         }
 
+        [Fact]
+        public static void Regression38565_Serialize()
```

Comment:
I am not a fan of putting issue numbers in the code like this and naming classes etc after issues. Do we do this elsewhere?

Reply:
Other codebases, such as CoreCLR or Roslyn frequently do. I'm unsure about corefx in particular.

If you have an idea about better/more meaningful names, I'm all for it, but `Regression####` clearly covers that this is a regression test for a particular issue, avoids conflicts with anything else. It also indicates that it probably shouldn't be modified or reused elsewhere and avoids having to come up with some "meaningful" name for a test (which is realistically unimportant).

The only thing it doesn't capture is that the bug id came from github (which Roslyn covers using attributes).

Reply:
For a very specific or edge case test I can see putting an issue in the description, but I probably still wouldn't since we have history available and the tests should be self-descriptive IMO. The "null" tests represent basic support so, for example, the existing tests should be extended if there are some that already tests other collection types.

Reply:
Having basic tests is fine, but I think having explicit regression tests, which are called out as such, is just as important. Having a test that was created explicitly due to a bug and being able to see that without traversing history is important, as it helps show cases that were originally missed and ensures that refactorings/rewrites don't remove them.

It looks like in the CoreCLR case they label them `GitHub#####` rather than `Regression#####`, which covers the missing bit of where the bug came from for scenarios where source control or the bug tracking database changes.

Reply:
Every test is a "regression" test. This area is under active development so why should we treat this as a "legacy" code base by having a bunch of one-off tests that make the tests harder to maintain and understand?

Reply:
> so why should we treat this as a "legacy" code base by having a bunch of one-off tests that make the tests harder to maintain and understand?

I don't think it is treating it as legacy code. It is just handling explicit customer bugs, tracking that this test was added because of a missed scenario, that the code being tested was given by a customer or is directly related to a customer feedback.

Reply:
Let's discuss offline.

Reply:
FWIW, I agree with @steveharter that it doesn't make sense to use non-descript test names. Adding a pointer to the customer bug can be addressed by adding a comment to the issue itself (and keep the test name descriptive).

I think we can use "Regression" labels for test names after shipping, especially when it comes to servicing branches.

However, I would also rather get the fix in quickly since this to me is just a non-blocking nit :)